### PR TITLE
Support authenticate with SecurityTokenService for Lambda application deployment

### LINKED ISSUE #1515: Support authenticate with SecurityTokenService
**What would you like to be added**:

Be able to authenticate piped agent with service account role from EKS pod.

**Why is this needed**:

Currently, we only support authentication in serval ways as below
1. from the environment variables.
2. from the given credentials file.
3. from the EC2 Instance Role.

But in case of the piped agent running in a pod on EKS cluster, it can't get the service account role (of running EC2 node) and use for authentication.

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1515

**Does this PR introduce a user-facing change?**:

```release-note
Support authenticate with SecurityTokenService for Lambda application deployment
```

### CODE REVIEW

**Comment by nakabonne** [comment]
I feel we should update this page as well.
https://pipecd.dev/docs/operator-manual/piped/adding-a-cloud-provider/#configuring-lambda-cloud-provider

**Comment by nakabonne** [comment]
Also, here as well: https://github.com/pipe-cd/pipe/blob/363d89c6a837878d1a66118eaaae8b9653352e1f/pkg/config/piped.go#L314-L319

**Comment by khanhtc1202** [comment]
> Also, here as well:
> 
> https://github.com/pipe-cd/pipe/blob/363d89c6a837878d1a66118eaaae8b9653352e1f/pkg/config/piped.go#L314-L319

I just remove the previous description to [pkg/app/piped/cloudprovider/lambda/client.go](https://github.com/pipe-cd/pipe/pull/1516/files#diff-b0e2109e8d4bd99f3b077c85e30da6b8568ef2dbd005099ae075406775cb4f68) since the type `CloudProviderLambdaConfig` is definition side, I think adding this description to using side and the docs is enough 😅

**Comment by khanhtc1202** [comment]
> I feel we should update this page as well.
> https://pipecd.dev/docs/operator-manual/piped/adding-a-cloud-provider/#configuring-lambda-cloud-provider

Thanks, nice catch 👍 
Addressed on https://github.com/pipe-cd/pipe/pull/1516/commits/1081f75483bfb6e30a560045f2ed53411e4c8494 🙏

File: docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md [hunk 1]
```diff
@@ -97,7 +97,8 @@ You will generally need your AWS credentials to authenticate with Lambda. Piped
 It attempts to retrieve credentials in the following order:
 1. From the environment variables. Available environment variables are `AWS_ACCESS_KEY_ID` or `AWS_ACCESS_KEY` and `AWS_SECRET_ACCESS_KEY` or `AWS_SECRET_KEY`.
 2. From the given credentials file. (the `credentialsFile field in above sample`)
-3. From the EC2 Instance Role.
+3. from the pod running in EKS cluster via STS (SecurityTokenService).
```
Comment (nghialv) [inline docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md:100]: nit: "From"

Comment (khanhtc1202) [reply docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md]: addressed on https://github.com/pipe-cd/pipe/pull/1516/commits/5a7f7d771d112e8363e00034810a0c18344707f2 🙏

**Comment by nakabonne** [comment]
Thank you.
/lgtm

**Comment by nghialv** [comment]
> Add a new way to authenticate piped agent by service account role from EKS pod

Because the NOTE will be added to ChangeLog when we release this PR, so can you change this to tell users what way was added.
The PR title looks good.

**Comment by khanhtc1202** [comment]
> > Add a new way to authenticate piped agent by service account role from EKS pod
> 
> Because the NOTE will be added to ChangeLog when we release this PR, so can you change this to tell users what way was added.
> The PR title looks good.

Sure, updated 👍 Thank you 🙏

**Comment by nghialv** [comment]
Nice.
/approve

**Pull request merged**